ldms/src/sampler/zfs_leafvdevs/zfs_leafvdevs.c

```diff
@@ -302,15 +304,19 @@ struct ldmsd_plugin *get_plugin(ldmsd_msg_log_f pf)
  *  init_errors
  */
 static int get_leafvdev_stats(nvlist_t *nvroot, const char *pool_name,
-    const char *parent_name)
+    const char *parent_name, zpool_handle_t *zhp)
 {
         uint_t c;
         vdev_stat_t *vs;
         char *vdev_name = NULL;
+	char *vdev_path = NULL;
+	char *vdev_guid = NULL;
 	ldms_mval_t record_instance;
         int rc=0; /*return code*/
 
-	vdev_name = get_vdev_name(nvroot, parent_name);
```

Comment:
All instances `get_vdev_name` should be replaceable with `zpool_vdev_name`, then you could remove this function.